### DISCUSSION TITLE
dynamic trigger form validation on submit

### DISCUSSION THREAD

**Discussion by tafelito** [react-hook-form/react-hook-form#2454] [Q&A]
What would be the best way without using setSate to conditionally trigger validation on submit?

I have a form that has a Save and a Done button. The save should submit the form without triggering the validation and the done should instead do the validation on submit

The validation is done with yup

Thanks!

**Comment by bluebill1049** [comment]
can you use context in the resolver to switch on and off with validation? or perhaps just use `getValues()` and read the entire form values.

**Comment by tafelito** [comment]
I'm using yupResolver as the resolver. I'd have to use a custom resolver and check the values and decide whether or not to call the yupResolver there or not. I'm trying to find a way to use the yupResolver inside a custom resolver but I having some TS issues

**Comment by tafelito** [comment]
I had to do something like this

```js
resolver: async (values, context, validateAllFieldCriteria) => {
      if (values.done) {
        const newContext = { ...context, someCondition: true },
        const resolver = yupResolver(schema);
        const res = await resolver(values, newContext, validateAllFieldCriteria);
        return res as any;
      }
      return {
        values,
        errors: {}
      }
    }, 
```

That seems to work but I still need a way to prevent the form to submit when I press enter in any input inside the form
